### PR TITLE
Added Bcomp Fee Codes and Tests

### PR DESCRIPTION
**Pull request by cameron-freshworks**
*Issue #:* /bcgov/entity#4494

*Description of changes:*

* Added Conditionals and Fee Codes for BCOMP Maintenance Filings
* Updated Unit testing for Fee Codes

*Notes: *

* My exposure to the legal api has been limited, so please be critical and don't hesitate to mention any implications these changes or approach might have throughout. Thanks!

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of the lear license (Apache 2.0).

### CODE REVIEW

**Comment by cameron-freshworks** [comment]
> Hey Cameron, this looks good to me! I'm just wondering if the fee code for bcomp annual reports should also be included here? It's not in your ticket, but I don't think we are covering it anywhere else so maybe it was missed in planning and should be added in here?

Great question, i will bring it up at stand up. I didn't see it specified in either the UI or Legal Api tickets, my only assumption is that it wasn't a priority for Relationships or us due to the time frame of when a user will actually start filing AR's. It definitely makes sense to update that code though, just depends if that code actually exists on Relationships side yet :)

**Comment by cameron-freshworks** [comment]
@kialj876 Will be updating BCOMP Ar Code to "BCANN" as per a follow up with Sumesh 👍

**Comment by cameron-freshworks** [comment]
@kialj876 Updated 👍

File: legal-api/tests/unit/api/test_business_filings/test_filings.py [hunk 1]
```diff
@@ -785,20 +787,37 @@ def test_get_correct_fee_codes(session):
     for director in free_cod['filing']['changeOfDirectors']['directors']:
         if not all(action in ['nameChanged', 'addressChanged'] for action in director.get('actions', [])):
             director['actions'] = ['nameChanged', 'addressChanged']
+    bc_coa = copy.deepcopy(FILING_HEADER)
+    bc_coa['filing']['header']['name'] = 'changeOfAddress'
+    bc_coa['filing']['business']['legalType'] = 'BC'
+    bc_coa['filing']['changeOfAddress'] = CHANGE_OF_ADDRESS
+    bc_cod = copy.deepcopy(FILING_HEADER)
+    cod['filing']['header']['name'] = 'changeOfDirectors'
```
Comment (thorwolpert) [inline legal-api/tests/unit/api/test_business_filings/test_filings.py:795]: duplicate of line 779, should it be bc_cod?

Comment (cameron-freshworks) [reply legal-api/tests/unit/api/test_business_filings/test_filings.py:795]: It should indeed! Updated in next commit

**Pull request merged**